File: 30 Days of Code/Java/Day19_Interfaces.java
import java.io.*;
import java.util.*;

interface AdvancedArithmetic{
   int divisorSum(int n);
}
//import java.math.*;
class Calculator implements AdvancedArithmetic {
    public int divisorSum(int n) {
        int sum = 0;
        for(int i=1; i<=(int)Math.sqrt(n); i++){
            if(n%i == 0){
                if(i == (n/i))sum = sum + i;
                else sum = sum + (i + n/i);
            }
        }
        
        
        return sum;
    }
}

class Solution {

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int n = scan.nextInt();
        scan.close();
        
      	AdvancedArithmetic myCalculator = new Calculator(); 
        int sum = myCalculator.divisorSum(n);
        System.out.println("I implemented: " + myCalculator.getClass().getInterfaces()[0].getName() );
        System.out.println(sum);
    }
}

File: 30 Days of Code/Python/Day19_Interfaces.py
class AdvancedArithmetic(object):
    def divisorSum(n):
        raise NotImplementedError
import math
class Calculator(AdvancedArithmetic):
    def divisorSum(self, n):
        l = []
        for i in range(1, int(math.sqrt(n)+1)):
            if n%i == 0:
                if i == (n/i):
                    l.append(i)
                else: 
                    l.append(i)
                    l.append(n/i)
        return int(sum(l))
                
                    


n = int(input())
my_calculator = Calculator()
s = my_calculator.divisorSum(n)
print("I implemented: " + type(my_calculator).__bases__[0].__name__)
print(s)